controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -253,13 +253,13 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&ocsv1.OCSInitialization{}).
-		Owns(&corev1.Service{}).
-		Owns(&corev1.Secret{}).
-		Owns(&promv1.Prometheus{}).
-		Owns(&corev1.ConfigMap{}).
-		Owns(&promv1.Alertmanager{}).
-		Owns(&promv1.ServiceMonitor{}).
+		For(&ocsv1.OCSInitialization{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
this controller is for ocsinit, I don't think this should be filtered via predicate.

Reply:
It should be otherwise it will trigger 2 reconcile for the same thing, one for the actual change and one for the status update in that reconcile.

Reply:
unless ocsinit status has timestamps or not ordered arrays among other things, a single extra reconcile is fine as status for a desired state is almost constant.

Reply:
Here is a yaml of it.
```apiVersion: v1
items:
- apiVersion: ocs.openshift.io/v1
  kind: OCSInitialization
  metadata:
    creationTimestamp: "2024-05-21T14:12:19Z"
    generation: 1
    name: ocsinit
    namespace: openshift-storage
    resourceVersion: "157119"
    uid: 4537ce40-5856-4418-9599-c75987b71511
  spec: {}
  status:
    conditions:
    - lastHeartbeatTime: "2024-05-21T14:26:07Z"
      lastTransitionTime: "2024-05-21T14:12:19Z"
      message: Reconcile completed successfully
      reason: ReconcileCompleted
      status: "True"
      type: ReconcileComplete
    - lastHeartbeatTime: "2024-05-21T14:26:07Z"
      lastTransitionTime: "2024-05-21T14:12:19Z"
      message: Reconcile completed successfully
      reason: ReconcileCompleted
      status: "True"
      type: Available
    - lastHeartbeatTime: "2024-05-21T14:26:07Z"
      lastTransitionTime: "2024-05-21T14:12:19Z"
      message: Reconcile completed successfully
      reason: ReconcileCompleted
      status: "False"
      type: Progressing
    - lastHeartbeatTime: "2024-05-21T14:26:07Z"
      lastTransitionTime: "2024-05-21T14:12:19Z"
      message: Reconcile completed successfully
      reason: ReconcileCompleted
      status: "False"
      type: Degraded
    - lastHeartbeatTime: "2024-05-21T14:26:07Z"
      lastTransitionTime: "2024-05-21T14:12:19Z"
      message: Reconcile completed successfully
      reason: ReconcileCompleted
      status: "True"
      type: Upgradeable
    phase: Ready
    rookCephOperatorConfig: {}
    sCCsCreated: true
kind: List
metadata:
  resourceVersion: ""
```

Reply:
> unless ocsinit status has timestamps
- based on above, it has, can be closed.